lab3/client/src/reducer.js

```diff
@@ -0,0 +1,45 @@
+import constants from './constants';
+
+var index = 0;
+
+export default function reducer (state, action) {
+	switch (action.type) {
+	case constants.actions.EXAMPLE:
+		state.example = action.payload;
+		return state;
+	case constants.actions.BUTTON_CLICK:
+		state.counter ++;
+		return state;	
+	case constants.actions.BUY_GENERATOR:
+		const currentGenerator = action.generatorClicked || 0;
```

Comment:
This will not pass the unit test. Please check on what unit test passes as action.